### ISSUE TITLE
boolean Helm parameters in Application CRD make UI/CLI unusable

### ISSUE THREAD

**Issue by stevesea** [argoproj/argo-cd#1551]
**Describe the bug**
If i use a boolean `true` instead of string `"true"` as the value of a Helm parameter, the UI and CLI both become unusable.  (version 0.12.2)

error looks like:
```
> argocd app list
FATA[0000] rpc error: code = Unknown desc = v1alpha1.ApplicationList.Items: []v1alpha1.Application: v1alpha1.Application.Spec: v1alpha1.ApplicationSpec.Source: v1alpha1.ApplicationSource.Helm: v1alpha1.ApplicationSourceHelm.Parameters: []v1alpha1.HelmParameter: v1alpha1.HelmParameter.Value: ReadString: expects " or n, but found t, error found in #10 byte of ...|,"value":true},{"nam|..., bigger context ...|v"},{"name":"default.app.enableAutoSync","value":true},{"name":"applications.APPNAME.enableAutoSyn|... 
```

I couldn't workaround the problem by pushing changes to the Application CRD in git. To ArgoCD back into a happy state, I used `kubectl edit application` to manually fix the offending CRDs

**To Reproduce**
To reproduce, create an Application CRD that has boolean values. For example: 
```
source:
  helm:
    parameters:
      - name: developerName
        value: dev
      - name: janitorTTL
        value: 30d
      - name: autoSync.apps.umbrella
        value: true
      - name: autoSync.infra.umbrella
        value: true
      - name: autoSync.apps.appDefault
        value: true
      - name: autoSync.infra.appDefault
        value: true
```

**Expected behavior**
Validation errors or warnings in the UI/CLI when the invalid CRD is submitted.

**Comment by alexec** [comment]
Have you tried quoting the `true`, e.g.

```yaml
 - name: autoSync.infra.appDefault
        value: "true"
```

**Comment by stevesea** [comment]
Yes, quoting the "true" to make it a String works. 

I logged an issue since I had no recourse through the UI/CLI to find / fix the offending Application CRDs. Also, the error messages didn't direct me to a particular Application CRD that needed to be fixed, and both UI/CLI were put into a state where I couldn't use them to manage the Applications that didn't have problems.

**Comment by alexec** [comment]
So the bug is that you can only fix this by editting the CM via `kubectl` but you cannot fix this via UI or CLI?

**Comment by alexmt** [comment]
The protection from malformed applications is available in v1.0-rc (implemented by https://github.com/argoproj/argo-cd/issues/1245). Now CRD definition contains schema definition and kubernetes rejects field if spec is invalid. Closing this one as duplicate
